### ISSUE TITLE
Is `xlsxwriter` installed during SCT installation?

### ISSUE THREAD

**Issue by valosekj** [spinalcordtoolbox/spinalcordtoolbox#4253]
I experienced missing `xlsxwriter` python package on two machines (my laptop and joplin) when calling `sct_analyze_lesion`.

<details><summary>log</summary>

```console
+ sct_analyze_lesion -m sub-5546_T2w_lesion-manual_bin.nii.gz -s sub-5546_T2w_seg-manual.nii.gz -ofolder /home/GRAMES.POLYMTL.CA/p118175/extrassd1/janvalosek/sci-multisite_analyze_lesions_seed50/results

--
Spinal Cord Toolbox (git-master-8a49d2949a3bf2affb56501be7cffae077a134e6)

sct_analyze_lesion -m sub-5546_T2w_lesion-manual_bin.nii.gz -s sub-5546_T2w_seg-manual.nii.gz -ofolder /home/GRAMES.POLYMTL.CA/p118175/extrassd1/janvalosek/sci-multisite_analyze_lesions_seed50/results
--

Creating temporary folder (/tmp/sct_2023-10-11_08-09-57_analyze-lesion_tb3ix9ec)
cp sub-5546_T2w_lesion-manual_bin.nii.gz /tmp/sct_2023-10-11_08-09-57_analyze-lesion_tb3ix9ec
cp sub-5546_T2w_seg-manual.nii.gz /tmp/sct_2023-10-11_08-09-57_analyze-lesion_tb3ix9ec

Label connected regions of the masked image...
^[[33mImage header specifies datatype 'uint8', but array is of type 'int64'. Header metadata will be overwritten to use 'int64'.^[[0m
Lesion count = 1

Measures on lesion #1...
  (S-I) length : 12.7 mm
  Max. equivalent diameter : 7.56 mm
  Maximum axial damage ratio : 0.51
  Volume : 489.74 mm^3


Averaged measures...
  volume [mm3] = 489.74 +/- 0.0
  length [mm] = 12.7 +/- 0.0
  max_equivalent_diameter [mm] = 7.56 +/- 0.0
  max_axial_damage_ratio [] = 0.51 +/- 0.0

Total volume = 489.74 mm^3
Lesion count = 1
Traceback (most recent call last):
  File "/home/GRAMES.POLYMTL.CA/p118175/code/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_analyze_lesion.py", line 644, in <module>
    main(sys.argv[1:])
  File "/home/GRAMES.POLYMTL.CA/p118175/code/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_analyze_lesion.py", line 620, in main
    lesion_obj.analyze()
  File "/home/GRAMES.POLYMTL.CA/p118175/code/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_analyze_lesion.py", line 184, in analyze
    self.pack_measures()
  File "/home/GRAMES.POLYMTL.CA/p118175/code/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_analyze_lesion.py", line 199, in pack_measures
    writer = pd.ExcelWriter(self.excel_name, engine='xlsxwriter')
  File "/home/GRAMES.POLYMTL.CA/p118175/code/spinalcordtoolbox/python/envs/venv_sct/lib/python3.8/site-packages/pandas/io/excel/_xlsxwriter.py", line 184, in __init__
    from xlsxwriter import Workbook
ModuleNotFoundError: No module named 'xlsxwriter'
```

</details>

Once I installed `xlsxwriter` manually into the SCT venv by the following commands, `sct_analyze_lesion` works fine.

```console
$ source ${SCT_DIR}/python/etc/profile.d/conda.sh
$ conda activate venv_sct
$ pip install xlsxwriter
``` 

I am wondering if `xlsxwriter` is installed during SCT installation. If so, why was it not installed for my SCT installations?

**Comment by mguaypaq** [comment]
It seems to be installed since earlier this year (https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/25f9a6b14a7e0f82e210d8e68565a69f7420a09d):
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/8a49d2949a3bf2affb56501be7cffae077a134e6/requirements.txt#L52-L54

It looks like this dependency is included in SCT v6.0, but not in SCT v5.8 ([`requirements-freeze.txt`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b8be5c4debcb2cbe3a02636d7a0b78172b411876/requirements-freeze.txt)) or earlier. But you're running from the latest `master` commit, so it should be fine.

Possibly what happened is that you had a development version (from the `master` branch) installed before `requirements.txt` got updated, and then did a `git pull` to a newer version? This doesn't automatically upgrade or install new dependencies with `pip`. But any fresh installs of SCT should have the right dependencies installed.

If your two experiences are indeed with long-lived development versions of SCT, I think we can close this issue (and re-open if it happens again). Otherwise we should investigate some more.

**Comment by valosekj** [comment]
Thanks for the investigation! Yeah, I guess both my SCT installations were installed a while ago (i.e. before requirements.txt got updated).
